lib/cisco_node_utils/vtp.rb

```diff
@@ -81,18 +69,11 @@ def password=(password)
       fail TypeError if password.nil?
       fail TypeError unless password.is_a? String
       fail ArgumentError if password.length > MAX_VTP_PASSWORD_SIZE
-      enable unless Vtp.enabled
-      begin
-        if password == default_password
-          config_set('vtp', 'password', 'no', '')
-        else
-          config_set('vtp', 'password', '', password)
-        end
-      rescue Cisco::CliError => e
-        raise unless e.clierror =~ /password cannot be set for NULL domain/
-        unless password == default_password
-          raise 'Setting VTP password requires first setting VTP domain'
-        end
+      Feature.vtp_enable
+      if password == default_password
+        config_set('vtp', 'password', state: 'no', password: '')
+      else
+        config_set('vtp', 'password', state: '', password: password)
```

Comment:
Style Nit: I prefer a single `config_set` call. Since default_password is `''` you could just do:

```
    def password=(password)
      Feature.vtp_enable
      state = (password == default_password) ? 'no' : ''
      config_set('vtp', 'password', state: 'no', password: '')
   end
```